### PR TITLE
Refs #23919 -- Removed six.PY2/PY3 usage

### PR DESCRIPTION
**Pull request by claudep**

### CODE REVIEW

File: django/contrib/auth/migrations/0001_initial.py [hunk 1]
```diff
@@ -64,7 +64,7 @@ class Migration(migrations.Migration):
                     help_text='Required. 30 characters or fewer. Letters, digits and @/./+/-/_ only.', unique=True,
                     max_length=30, verbose_name='username',
                     validators=[
-                        validators.UnicodeUsernameValidator() if six.PY3 else validators.ASCIIUsernameValidator()
+                        validators.UnicodeUsernameValidator()
```
Comment (timgraham) [inline django/contrib/auth/migrations/0001_initial.py]: single line would be consistent with the other migrations

File: django/core/handlers/wsgi.py [hunk 2]
```diff
@@ -212,22 +211,20 @@ def get_bytes_from_wsgi(environ, key, default):
     """
     Get a value from the WSGI environ dictionary as bytes.
 
-    key and default should be str objects. Under Python 2 they may also be
-    unicode objects provided they only contain ASCII characters.
+    key and default should be str objects.
     """
     value = environ.get(str(key), str(default))
-    # Under Python 3, non-ASCII values in the WSGI environ are arbitrarily
+    # Non-ASCII values in the WSGI environ are arbitrarily
```
Comment (timgraham) [inline django/core/handlers/wsgi.py]: could you rewrap the comment?

File: django/db/models/base.py [hunk 3]
```diff
@@ -510,8 +510,6 @@ def __repr__(self):
         return force_str('<%s: %s>' % (self.__class__.__name__, u))
 
     def __str__(self):
-        if six.PY2 and hasattr(self, '__unicode__'):
-            return force_text(self).encode('utf-8')
         return str('%s object' % self.__class__.__name__)
```
Comment (timgraham) [inline django/db/models/base.py]: is `str()`still needed?

File: django/db/models/manager.py [hunk 4]
```diff
@@ -87,8 +86,7 @@ def manager_method(self, *args, **kwargs):
 
         new_methods = {}
         # Refs http://bugs.python.org/issue1785.
```
Comment (timgraham) [inline django/db/models/manager.py]: Did you check if this comment is still relevant? If so, a better explanation would be useful, though not needed as part of this PR.

**Comment by claudep** [comment]
@timgraham I'll fix the isort issue when pushing, unless you want all greens before reviewing

File: django/dispatch/dispatcher.py [hunk 5]
```diff
@@ -2,15 +2,9 @@
 import threading
 import weakref
 
-from django.utils import six
 from django.utils.inspect import func_accepts_kwargs
 from django.utils.six.moves import range
 
-if six.PY2:
-    from .weakref_backports import WeakMethod
```
Comment (timgraham) [inline django/dispatch/dispatcher.py]: Please remove `weakref_backports.py` and its mention in `setup.cfg`.

File: django/http/cookie.py [hunk 6]
```diff
@@ -1,20 +1,15 @@
...
-from django.utils.encoding import force_str
 from django.utils.six.moves import http_cookies
 
-# http://bugs.python.org/issue2193 is fixed in Python 3.3+.
-_cookie_allows_colon_in_names = six.PY3
-
 # Cookie pickling bug is fixed in Python 2.7.9 and Python 3.4.3+
 # http://bugs.python.org/issue22775
 cookie_pickles_properly = (
     (sys.version_info[:2] == (2, 7) and sys.version_info >= (2, 7, 9)) or
```
Comment (timgraham) [inline django/http/cookie.py]: Want to remove the Python 2.7 stuff here?

File: django/test/client.py [hunk 7]
```diff
@@ -322,10 +322,10 @@ def _get_path(self, parsed):
         if parsed[3]:
             path += str(";") + force_str(parsed[3])
         path = uri_to_iri(path).encode(UTF_8)
-        # Under Python 3, non-ASCII values in the WSGI environ are arbitrarily
+        # Non-ASCII values in the WSGI environ are arbitrarily
```
Comment (timgraham) [inline django/test/client.py]: rewrap and reword: "Replace the behavior where Non-ASCII values in the WSGI environ are arbitrarily ..."

File: django/test/utils.py [hunk 8]
```diff
@@ -615,7 +610,7 @@ def is_quoted_unicode(s):
 
 
 def str_prefix(s):
```
Comment (timgraham) [inline django/test/utils.py]: Do you have it on your todo to remove this function?

File: django/utils/datastructures.py [hunk 9]
```diff
@@ -196,23 +196,9 @@ def _itervalues(self):
...
-
-        def items(self):
-            return list(self.iteritems())
-
-        def lists(self):
-            return list(self.iterlists())
-
-        def values(self):
-            return list(self.itervalues())
+    items = _iteritems
```
Comment (timgraham) [inline django/utils/datastructures.py]: Any problem renaming the underscore versions instead?

File: django/utils/glob.py [hunk 10]
```diff
@@ -1,19 +1,2 @@
...
-    _magic_check = re.compile('([*?[])')
-
-    def glob_escape(pathname):
-        """
-        Escape all special characters.
-        """
-        drive, pathname = os.path.splitdrive(pathname)
-        pathname = _magic_check.sub(r'[\1]', pathname)
-        return drive + pathname
+# To be deprecated
```
Comment (timgraham) [inline django/utils/glob.py]: I think that could be removed without deprecation since it's undocumented and only used once in Djagno.

File: django/utils/html_parser.py [hunk 11]
```diff
@@ -8,15 +7,13 @@
...
-        argument.
-        """
-        def __init__(self, convert_charrefs=False, **kwargs):
-            _html_parser.HTMLParser.__init__(self, convert_charrefs=convert_charrefs, **kwargs)
-else:
-    HTMLParser = _html_parser.HTMLParser
+class HTMLParser(_html_parser.HTMLParser):
+    """Explicitly set convert_charrefs to be False.
+
+    This silences a deprecation warning on Python 3.4, but we can't do
```
Comment (timgraham) [inline django/utils/html_parser.py]: remove "but we can't do...."

File: tests/template_tests/test_custom.py [hunk 12]
```diff
@@ -344,7 +342,6 @@ def test_load_working_egg(self):
             })
             engine.from_string(ttext)
 
-    @skipUnless(six.PY3, "Python 3 only -- Python 2 doesn't have annotations.")
     def test_load_annotated_function(self):
         Engine(libraries={
             'annotated_tag_function': 'template_tests.annotated_tag_function',
```
Comment (timgraham) [inline tests/template_tests/test_custom.py]: Remove `tests/template_tests/annotated_tag_function.py` from `setup.cfg`.

File: tests/view_tests/tests/test_debug.py [hunk 13]
```diff
@@ -29,9 +28,7 @@
     sensitive_args_function_caller, sensitive_kwargs_function_caller,
     sensitive_method_view, sensitive_view,
 )
-
-if six.PY3:
-    from .py3_test_debug import Py3ExceptionReporterTests  # NOQA
+from .py3_test_debug import Py3ExceptionReporterTests  # NOQA
```
Comment (timgraham) [inline tests/view_tests/tests/test_debug.py]: Remove `py3_test_debug.py` from setup.cfg (a better solution here is to move the test into this file).

File: django/db/models/manager.py (hunk 4 above)
Comment (claudep) [inline django/db/models/manager.py]: This could be added to #23919 tasks.

File: django/http/cookie.py (hunk 6 above)
Comment (claudep) [inline django/http/cookie.py]: I'd rather separate concerns and do that in another PR.

File: django/db/models/manager.py (hunk 4 above)
Comment (timgraham) [inline django/db/models/manager.py]: I think it's related to the `predicate = inspect.isfunction if six.PY3 else inspect.ismethod` line and can be removed (31fadc120213284da76801cc7bc56e9f32d7281b).

File: django/test/utils.py (hunk 8 above)
Comment (claudep) [inline django/test/utils.py]: Not yet, one thing at a time :-)

File: django/utils/html_parser.py (hunk 11 above)
Comment (claudep) [inline django/utils/html_parser.py]: I left the comment as this would need examination whether some more removal should be done (we'll grep for 2.7 in a future patch).

File: django/db/models/manager.py (hunk 4 above)
Comment (claudep) [inline django/db/models/manager.py]: Ah yes, sorry. I'll remove the comment.

**Comment by claudep** [comment]
Oh, yes... sorry!
